mod/cli/pkg/commands/deposit/create.go

```diff
@@ -139,55 +151,167 @@ func createValidatorCmd(
 			"signature", signature.String(),
 		)
 
-		// TODO: once broadcast is fixed, remove this.
-		logger.Info("Send the above calldata to the deposit contract 🫡")
+		if broadcast {
+			var txHash common.ExecutionHash
+			txHash, err = broadcastDepositTx[ExecutionPayloadT](
+				cmd, depositMsg, signature, chainSpec,
+			)
+			if err != nil {
+				return err
+			}
+
+			logger.Info(
+				"Deposit transaction successful",
+				"txHash", txHash.Hex(),
+			)
+		}
 
 		return nil
 	}
 }
 
+func broadcastDepositTx[
+	ExecutionPayloadT constraints.EngineType[ExecutionPayloadT],
+](
+	cmd *cobra.Command,
+	depositMsg *types.DepositMessage,
+	signature crypto.BLSSignature,
+	chainSpec common.ChainSpec,
+) (common.ExecutionHash, error) {
+	// Parse the private key.
+	var (
+		fundingPrivKey string
+		privKey        *ecdsa.PrivateKey
+	)
+	fundingPrivKey, err := cmd.Flags().GetString(privateKey)
+	if err != nil {
+		return common.ExecutionHash{}, err
+	}
+	if fundingPrivKey == "" {
+		return common.ExecutionHash{}, ErrPrivateKeyRequired
+	}
+	privKey, err = gethprimitivescrypto.HexToECDSA(fundingPrivKey)
+	if err != nil {
+		return common.ExecutionHash{}, err
+	}
+
+	if privKey == nil {
+		return common.ExecutionHash{}, ErrPrivateKeyEmpty
+	}
+
+	// Parse the execution client RPC URL.
+	rpcURL, err := cmd.Flags().GetString(rpcURL)
+	if err != nil {
+		return common.ExecutionHash{}, err
+	}
+	eth1ChainID := new(big.Int).SetUint64(chainSpec.DepositEth1ChainID())
+	depositContractAddress := chainSpec.DepositContractAddress()
+
+	// Dial the execution client.
+	rpcClient, err := rpc.DialContext(
+		cmd.Context(), rpcURL,
+	)
+	if err != nil {
+		return common.ExecutionHash{}, err
+	}
+	ethClient, err := ethclient.NewFromRPCClient[ExecutionPayloadT](
+		rpcClient,
+	)
+	if err != nil {
+		return common.ExecutionHash{}, err
+	}
+
+	// Send the deposit to the deposit contract through abi bindings.
+	depositContract, err := deposit.NewBeaconDepositContract(
+		depositContractAddress,
+		ethClient,
+	)
+	if err != nil {
+		return common.ExecutionHash{}, err
+	}
+	depositTx, err := depositContract.Deposit(
+		&bind.TransactOpts{
+			From: gethprimitivescrypto.PubkeyToAddress(privKey.PublicKey),
+			Signer: func(
+				_ common.ExecutionAddress, tx *gethprimitives.Transaction,
+			) (*gethprimitives.Transaction, error) {
+				return gethprimitives.SignTx(
+					tx,
+					gethprimitives.LatestSignerForChainID(eth1ChainID),
+					privKey,
+				)
+			},
+			//nolint:mnd // The gas tip cap.
+			// It is necessary for besu to work, not sure why though.
```

Comment:
remove these comments

also the reason why they're required is the client is just not inferring a valid set of gas parameters so we need to specify it manually

Reply:
hehe, removed :D Thanks!